base/channels.jl

```diff
@@ -189,6 +189,39 @@ function check_channel_state(c::Channel)
         throw(closed_exception())
     end
 end
+
+function _do_is_closed(f)
+    # This function will run `f()` and return `true` if it throws an `InvalidStateException` because
+    # a channel is closed, and `false` otherwise.
+    try
+        f()
+        return false
+    catch e
+        if isa(e, InvalidStateException) && e.state === :closed
+            return true
+        else
+            rethrow()
+        end
+    end
+end
+
+function _positive_int(x::T, purpose::String) where {T<:Union{AbstractFloat, Integer}}
```

Comment:
Currently, calling `Channel(1.5)` throws an InexactError trying to convert to Int. However, the ArgumentError text already captures the real issue (size must be 0, a positive int, or Inf), so I think the additional error type is unnecessary and a bit confusing in the stacktrace (it doesn't point to the issue being with Channel).

So I wrote this check and I'm using it to check to make sure `n` is always positive. If people agree with this design choice, it could be extended to the Channel constructor in the future.